### ISSUE TITLE
Add `fill_null` on dataframe/expr level

### ISSUE THREAD

**Issue by ion-elgreco** [apache/datafusion#12906]
### Is your feature request related to a problem or challenge?

I would like to be able to fill_nulls per col/expr and on the dataframe level, akin to polars/pyspark/pandas

### Describe the solution you'd like

Add a fill_null method to DataFrame struct, add fill_null logical expr.

**Comment by Omega359** [comment]
This seems like it may almost be an alias for [nvl](https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#nvl) or [nvl2](https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html#nvl2)

**Comment by ion-elgreco** [comment]
@Omega359 indeed, for now I've opened a PR for a fill_null in datafusion-python that calls nvl: https://github.com/apache/datafusion-python/pull/919
